File: 1951-find-the-winner-of-the-circular-game/find-the-winner-of-the-circular-game.java
class Solution {
    public int findTheWinner(int n, int k) {
        Queue<Integer> q=new LinkedList<>();
        for(int i=1;i<=n;i++){
            q.add(i);
        }
        int c=0;
        while(q.size()>1){
            c++;
            int r=q.poll();
            if(c==k){
                // q.poll();
                c=0;
            }
            else{
                q.add(r);
            }
        }
        return q.peek();
    }
}

File: 1951-find-the-winner-of-the-circular-game/find-the-winner-of-the-circular-game.cpp
class Solution {
public:
    int findTheWinner(int n, int k) {
        queue<int> q;
        int c=0;
        for(int i=1;i<=n;i++){
            q.push(i);
        }
        while(q.size()>1){
            int r=q.front();
            q.pop();
            c++;
            if(c==k){
                c=0;
            }
            else{
                q.push(r);
            }
        }
        return q.front();
    }
};